packages/webview_flutter/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -134,6 +133,68 @@ public void onProgressChanged(WebView view, int progress) {
     }
   }
 
+  /**
+   * Creates a {@link android.webkit.WebView} and configures it according to the supplied
+   * parameters.
+   *
+   * <p>The {@link WebView} is configured with the following predefined settings:
+   *
+   * <ul>
+   *   <li>always enable the DOM storage API;
+   *   <li>always allow JavaScript to automatically open windows;
+   *   <li>always allow support for multiple windows;
+   *   <li>always use the {@link FlutterWebChromeClient} as web Chrome client.
+   * </ul>
+   *
+   * <p><strong>Important:</strong> This method is visible for testing purposes only and should
+   * never be called from outside this class.
+   *
+   * @param webViewBuilder a {@link WebViewBuilder} which is responsible for building the {@link
+   *     WebView}.
+   * @param params creation parameters received over the method channel.
+   * @param webChromeClient an implementation of WebChromeClient This value may be null.
+   * @return The new {@link android.webkit.WebView} object.
+   */
+  @VisibleForTesting
+  static WebView createWebView(
+      WebViewBuilder webViewBuilder, Map<String, Object> params, WebChromeClient webChromeClient) {
```

Comment:
I agree, but then we should directly pass it into the `FlutterWebView` constructor. I can make that happen.

Reply:
Looking into this more detailed, changing this to a configuration object requires quite a large refactor as the `params` also contain a second `Map<String, Object>` collection containing several web settings. 

This means we need to change a lot of code that currently validates how to handle different situations (bases on if a key is part of the params or web settings hashmap or not). For now I think it would be better to leave it as is and maybe do a separate PR on updating this if needed.